src/NGen.cpp

```diff
@@ -754,8 +753,35 @@ int main(int argc, char* argv[]) {
 #endif
 
 #if NGEN_WITH_ROUTING
+#if NGEN_WITH_MPI
+    if (mpi_num_procs > 1) {
+        size_t flow_size = nexus_downstream_flows.size();
+        double *gather_downstream_flows = NULL;
+        if (mpi_rank == 0) {
+            gather_downstream_flows = (double *)calloc(flow_size * mpi_num_procs, sizeof(double));
+        }
```

Comment:
The partitions may not be perfectly equal in size right now, and I'd like to avoid baking such an assumption into this code.

Maybe there's something that already has the count of nexuses across the whole simulation. If not, it can be computed here with an `MPI_Reduce` call across processes.